make.bat

```diff
@@ -1,50 +1,154 @@
+    if /I "%~1" == "-clang" (
+        set compiler_opt="clang"
+        shift
+        goto :verifyopt
+    )
+) else (
+    echo A C compiler has already been specified: !compiler_opt!. 1>&2
+    exit /b 2
+)
+
+REM standard options
+if /I "%~1" == "-local" (
+    if !use_local! EQU 0 ( set /a use_local=1 )
+    shift
+    goto :verifyopt
+)
+if /I "%~1" == "-v" (
+    if !verbose_log! EQU 0 ( set /a verbose_log=1 )
+    shift
+    goto :verifyopt
+)
+if /I "%~1" == "--verbose" (
+    if !verbose_log! EQU 0 ( set /a verbose_log=1 )
+    shift
+    goto :verifyopt
+)
+if /I "%~1" == "-logfile" (
+    if /I "%~2" == "" (
+        echo Log file is not specified for -logfile parameter. 1>&2
+        exit /b 2
+    )
+    echo.>"%~sf2" || (
+        echo The log file specified for -logfile parameter does not exist. 1>&2
+        exit /b 2
+    )
+    set log_file="%~sf2"
+    shift
+    shift
+    goto :verifyopt
+)
 
 :init
-REM initialize the log file with the failure message
-set log_file=%TEMP%\v_make.bat.log
-echo Failed to compile - Create an issue at 'https://github.com/vlang/v' with the following info:>%log_file%
-echo.>>%log_file%
+del !log_file!>NUL 2>&1
+if !use_local! NEQ 1 (
+    if exist "vc" (
+        echo Updating vc...
+        cd vc
+        git pull --quiet
+        cd ..
+    ) else (
+        echo Cloning vc...
+        git clone --depth 1 --quiet https://github.com/vlang/vc.git
+    )
+)
 
-REM alleviate weird issues with this var
-set cloned_tcc=
-set ERRORLEVEL=
+echo Building V
 
-pushd %~dp0
+if !compiler_opt! EQU "clang" goto :clang_strap
+if !compiler_opt! EQU "gcc" goto :gcc_strap
+if !compiler_opt! EQU "msvc" goto :msvc_strap
+if !compiler_opt! EQU "tcc" goto :tcc_strap
+if !compiler_opt! EQU "fresh-tcc" goto :tcc_strap
+if !compiler_opt! EQU "" goto :clang_strap
 
-if "%~1"=="-local" goto :compile
-if "%~2"=="-local" goto :compile
+:clang_strap
+echo.
+echo Attempting to build v.c with Clang...
 
-if exist "vc" (
-	echo Updating vc...
-	cd vc
-	git pull --quiet
-	cd ..
-) else (
-	echo Cloning vc...
-	git clone --depth 1 --quiet https://github.com/vlang/vc
+where /q clang
+if %ERRORLEVEL% NEQ 0 (
+	echo  ^> Clang not found
+	if !compiler_opt! NEQ "" goto :error
+	goto :gcc_strap
 )
 
-:compile
-REM option to force msvc, gcc or tcc
-if "%~1"=="-gcc"        set force_gcc=1  & goto :gcc_strap
-if "%~2"=="-gcc"        set force_gcc=1  & goto :gcc_strap
-if "%~1"=="-msvc"       set force_msvc=1 & goto :msvc_strap
-if "%~2"=="-msvc"       set force_msvc=1 & goto :msvc_strap
-if "%~1"=="-tcc"        set force_tcc=1  & goto :tcc_strap
-if "%~2"=="-tcc"        set force_tcc=1  & goto :tcc_strap
-if "%~1"=="-fresh_tcc"  set force_tcc=1  & goto :fresh_tcc
-if "%~2"=="-fresh_tcc"  set force_tcc=1  & goto :fresh_tcc
+set /a valid_cc=1
+
+if !verbose_log! EQU 1 ( echo [Compile] clang -std=c99 -municode -pedantic -w -o v.exe .\vc\v_win.c>>!log_file! )
+clang -std=c99 -municode -pedantic -w -o v.exe .\vc\v_win.c>>!log_file! 2>>&1
```

Comment:
A lot of people won't have any version of clang installed.  They'll need to have tcc installed for them just to get this first compile to happen.

Reply:
if clang is not found in the PATH, the label will jump continuously until it reaches the tcc label, I thought clang is another promninent compiler so I included it

Reply:
No worries, then.  Yes, `cc` and `clang` are the most popular on other systems.  Just not on Windows.

Reply:
Without any flags passed, make.bat will do "auto detection" of avaliable C compilers(by unefficiently trying all the labels), if none were found, it will download TCC and display the `WARNING` message after the first compile.